core/state/state_object.go

```diff
@@ -91,6 +91,21 @@ type StateObject struct {
 	onDirty   func(addr common.Address) // Callback method to mark a state object newly dirty
 }
 
+// AccountObject is a reduced StateObject interface
+type AccountObject interface {
+	SubBalance(amount *big.Int)
+	AddBalance(amount *big.Int)
+	SetBalance(*big.Int)
+	SetNonce(uint64)
+	SetCode(common.Hash, []byte)
```

Comment:
Code should never be set. This is immutable. Once an account is created, its code is fixed until it suicides itself.